### ISSUE TITLE
bug(cdk/testing): Running e2e test with protractor throws error when using cdk/testing

### ISSUE THREAD

**Issue by Heines1983** [angular/components#24190]
### Is this a regression?

- [X] Yes, this behavior used to work in the previous version

### The previous version in which this bug was not present was

12

### Description

When running a e2e test with protractor using the following import statement an error is showed.

```ts
import { ComponentHarness, HarnessPredicate, BaseHarnessFilters } from "@angular/cdk/testing";
```

It seems that cdk/testing is somewhere using a require which is no longer allowed.

### Reproduction

Steps to reproduce:
1. Create a test which contains a `ComponentHarness` and/or `HarnessPredicate`.
2. Run the test.

### Expected Behavior

No error. The test runs successfully.

### Actual Behavior

```shell
Error: Error [ERR_REQUIRE_ESM]: require() of ES Module 
C:\git\ClientApp\node_modules\@angular\cdk\fesm2015\testing.mjs not supported.
```

Instead change the require of `C:\git\ClientApp\node_modules\@angular\cdk\fesm2015\testing.mjs` to a dynamic `import()` which is available in all CommonJS modules.

### Environment

- Angular:  13.1.2
- CDK/Material: 13.1.1
- Browser(s): Chrome 97
- Operating System (e.g. Windows, macOS, Ubuntu): Windows

**Comment by crisbeto** [comment]
It looks like a potential bundling issue. cc @devversion

**Comment by devversion** [comment]
It looks like the e2e spec files are executed in CommonJS mode (this was fine in Angular v12), but as of v13, all shipped Angular packages are strict ESM (hence the error in the issue)

@Heines1983 is this in an Angular CLI project or do you have a custom setup? some details here would be appreciated

**Comment by devversion** [comment]
Hey, thanks again for this issue. I brought this up with the team. 

Given that our Protractor harness environment is deprecated, together with `protractor` itself being deprecated, and the fact that very few people are impacted, we decided to not work on getting the harness environment working for CommonJS & Protractor as this would require conceptual changes in the Angular Package Format v13+.

On the other hand though, we are happy helping with finding a good strategy to migrate to `selenium-webdriver` which is officially maintained by the Selenium team. There is a harness environment available for that.

At a high-level, the migration to that should be fairly straightforward given that Protractor is actually based on `selenium-webdriver`. It will be mostly about changing imports, and waiting for elements to be visible in some situations (Protractor waited for Angular under the hood; this is not future-proof though and Cypress/other e2e solutions don't need that at all). I have created an example Angular CLI application that runs with `selenium-webdriver` and the select harness. https://github.com/angular/components/commit/aae62d16ed288f2f00308d942f38d8e40d9eedd1.

Closing this, but happy to continue disucssion!
